selfservice/strategy/oidc/provider_vk.go

```diff
@@ -106,8 +130,45 @@ func (g *ProviderVK) Claims(ctx context.Context, exchange *oauth2.Token, query u
 
 	user := response.Result[0]
 
-	if email, ok := exchange.Extra("email").(string); ok {
-		user.Email = email
+	if p.config.PKCE == "force" {
+		reqData := strings.NewReader("client_id=" + p.config.ClientID + "&access_token=" + exchange.AccessToken)
+		req, err = retryablehttp.NewRequestWithContext(ctx, "POST", "https://id.vk.com/oauth2/user_info", reqData)
+		if err != nil {
+			return nil, errors.WithStack(herodot.ErrInternalServerError.WithReasonf("%s", err))
+		}
+		req.Header.Set("Content-Type", "application/x-www-form-urlencoded")
+
+		resp, err = client.Do(req)
+		if err != nil {
+			return nil, errors.WithStack(herodot.ErrInternalServerError.WithReasonf("%s", err))
+		}
+		defer resp.Body.Close()
+
+		if err := logUpstreamError(p.reg.Logger(), resp); err != nil {
+			return nil, err
+		}
+
+		var userInfo struct {
+			User struct {
+				Email string `json:"email,omitempty"`
+				Phone string `json:"phone,omitempty"`
+			} `json:"user,omitempty"`
+		}
+
+		if err := json.NewDecoder(resp.Body).Decode(&userInfo); err != nil {
+			return nil, errors.WithStack(herodot.ErrInternalServerError.WithReasonf("%s", err))
+		}
+
+		if len(userInfo.User.Phone) > 0 && userInfo.User.Phone[0] != '+' {
+			userInfo.User.Phone = "+" + userInfo.User.Phone
```

Comment:
🤔 I'm not sure if this is the correct normalization to do here. I quickly checked VK docs, but couldn't find anything about the format.
If it is not clear, I'd prefer to do nothing instead and leave it to the consumer of the data.

Reply:
All I've got is two examples from the docs, as well as a confirmation that it works that way (I've tested on my VK account that is registered with a Russian phone number). I didn't find any detailed description of the format either.
The examples from the docs:
1. [Example 1](https://id.vk.com/about/business/go/docs/ru/vkid/latest/vk-id/connection/api-integration/api-description#Poluchenie-nemaskirovannyh-dannyh):
```jsonc
{
    "user": {
        "user_id": "1234567890", // Идентификатор пользователя.
        "first_name": "Ivan", // Имя пользователя.
        "last_name": "I.", // Первая буква фамилии пользвателя.
        "phone": "79903922415", // Телефон пользователя.
        "avatar": "https://pp.userapi.com/60tZWMo4SmwcploUVl9XEt8ufnTTvDUmQ6Bj1g/mmv1pcj63C4.png", // Ссылка на фото профиля пользователя.
        "email": "ivan_i123@vk.com", // Адрес почты пользователя.
        "sex": 2, // Пол. Возможные значения: Возможные значения: 1 — женский, 2 — мужской, 0 — пол не указан.
        "is_verified": false, // Признак того, что аккаунт пользователя подтвержден. Если это так, возвращается параметр true.
        "birthday": "01.01.2000" // Дата рождения пользователя.
    }
}
```
2. [Example 2](https://id.vk.com/about/business/go/docs/ru/vkid/latest/vk-id/connection/start-integration/auth-without-sdk/auth-without-sdk-web#Shag-8-Opcionalno-Poluchenie-polzovatelskih-dannyh):
```jsonc
{
    "user": {
        "user_id": "1234567890",
        "first_name": "Ivan",
        "last_name": "I.",
        "phone": "79991234567",
        "avatar": "http://avatar.com/12345678",
        "email": "ivan_i123@vk.com",
        "sex": 2,
        "verified": false,
        "birthday": "01.01.2000"
    }
}
```

`79903922415` and `79991234567` are Russian phone numbers in the international format, but without a leading `+` sign.
As far as I understand, Ory Kratos uses international format, but with a leading `+` sign.
I'm not sure whether all the phones are returned in the same format or not. For example I can't test with non-Russian phones. This is why I added condition `if there's no plus sign, then add it`.
If you insist, I can remove that code block (lines 162-164). It shouldn't be the problem to do the same action within a `.jsonnet` file though.